app/services/file_appender.rb

```diff
@@ -94,6 +94,7 @@ def create_file_set(file_node, file)
       )
       file_set = FileSet.new(attributes)
       change_set = ChangeSet.for(file_set)
+      change_set.validate(attributes)
```

Comment:
Does passing the attributes to new and then also to validate kind of set them twice?

Also I can't remember, did we add something in the change set persister that checks `valid?` before doing a save so that we can do things this way?

Reply:
> Does passing the attributes to new and then also to validate kind of set them twice?

It does set them twice - I originally only set it in `validate()`, but not every property that we set here is a ChangeSet property.

We don't check `valid?`, but we also didn't check that before either - this is mostly about making sure that the `files` virtual attribute gets through.